### PR TITLE
Refactor KVStore/RegionPersister test

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/5170

Problem Summary:
It is hard to write unit test cases for KVStore.
And it run tests on the KVStore instance in global context, which makes it hard to test logic of restoring data from disk

### What is changed and how it works?

* Explicitly send `PathPool` instance to `KVStore::restore` by param 
  * We can create a KVStore/RegionPersister instance for testing other than testing the KVStore instance in global context
  * Get a clearer order of initialization between PathPool and KVStore in Server.cpp 
* Rewrite the `RegionPersisterTest`
* Use DBMS_PUBLIC_GTEST instead of friend class to write tests on `RegionKVStoreTest`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by CalvinNeo — APPROVED** [review]
lgtm

File: dbms/src/Storages/Transaction/tests/gtest_kvstore.cpp [hunk 1]
```diff
@@ -36,47 +42,102 @@ extern void ChangeRegionStateRange(RegionState & region_state, bool source_at_le
...
+    static void testKVStoreRestore();
+
+protected:
+    static KVStore & getKVS() { return *kvstore; }
+    static KVStore & reloadKVSFromDisk()
+    {
+        kvstore.reset();
+        auto & global_ctx = TiFlashTestEnv::getGlobalContext();
+        kvstore = std::make_unique<KVStore>(global_ctx, TiDB::SnapshotApplyMethod::DTFile_Directory);
+        // TODO: Do we need to recreate proxy_instance and proxy_helper here?
```
Comment (CalvinNeo) [inline dbms/src/Storages/Transaction/tests/gtest_kvstore.cpp:94]: MockRaftStoreProxy do not persists its own raftstate for now, so we can't  recreate proxy for now.

Comment (JaySon-Huang) [reply dbms/src/Storages/Transaction/tests/gtest_kvstore.cpp]: updated

**Review by lidezhu — APPROVED** [review]

**Pull request merged**